Ginger/Ginger/Activities/ActivitiesGroupPage.xaml.cs

```diff
@@ -76,7 +79,131 @@ public ActivitiesGroupPage(ActivitiesGroup activitiesGroup, BusinessFlow parentB
                 SharedRepoInstanceUC.Visibility = Visibility.Collapsed;
                 SharedRepoInstanceUC_Col.Width = new GridLength(0);
             }
-        }     
+        }
+
+        void UpdateSharedRepositorySupportedOperations()
+        {
+            grdGroupedActivities.ShowUpDown = Visibility.Visible;
+
+            grdGroupedActivities.AddToolbarTool(Amdocs.Ginger.Common.Enums.eImageType.Add, "Add Another Activity to this Group", BtnAdd_Click);
+            grdGroupedActivities.AddToolbarTool(Amdocs.Ginger.Common.Enums.eImageType.DeleteSingle, "Delete Current Selected Activity from this Group", GroupActivitiesHandler);
+        }
+
+        private void BtnDelete_Click(object sender, RoutedEventArgs e)
+        {
+            FrameworkElement fElem = sender as FrameworkElement;
+            if (fElem != null)
+            {
+
+            }
+        }
+
+        private void BtnAdd_Click(object sender, RoutedEventArgs e)
+        {
+            ActivitiesRepositoryPage ActivitiesRepoPage = new ActivitiesRepositoryPage(WorkSpace.Instance.SolutionRepository.GetRepositoryItemRootFolder<Activity>(), null, null, GroupActivitiesHandler, true);
+            ActivitiesRepoPage.ShowAsWindow(Window.GetWindow(this));
+        }
+
+        private void GroupActivitiesHandler(object sender, RoutedEventArgs e)
+        {
```

Comment:
i didn't got for what GroupActivitiesHandler() was created and what it is doing.
it also uses IdentifiedActivity so for sure something is wrong with it.
please explain

Reply:
Done and Updated the code side and this handler calls for the ActivitiesRepository page to provide list of activities to add to the Shared Group